EVALUATION2.md

```diff
@@ -0,0 +1,84 @@
+## Hace todos los puntos pedidos (40%)
+
+#### Permite crear usuarios vía endpoint
+
+- No 😔
+
+#### Permite añadir favoritos vía endpoint
+
+OK
+
+#### Si el pokemon ya está marcado como favorito, ¿se lanza una excepción de dominio?
+
+OK
+
+#### Si el usuario no existe, ¿se lanza una excepción de dominio?
+
+- No
+
+#### Si se lanza una excepción desde el dominio, ¿se traduce en infraestructura a un código HTTP?
+
+OK
+
+#### Hay tests unitarios
+
+- Si, pero estos deberían estar separados del `src`. Es decir, por un lado debería estar el código de la aplicación
+  en `src` y a la altura de esta, debería haber otra carpeta `tests``
+
+**Error principal: No cumple los requisitos pedidos (no se pueden crear usuarios vía endpoint)**
+**Puntuación: 20/40**
+
+## Se aplican conceptos explicados (50%)
+
+#### Separación correcta de capas (application, domain, infrastructure + BC/module/layer)
+
+- El servicio de aplicación `AddFavoritePokemon` tiene dos responsabilidades: crear un usuario y añadir un pokemon
+  favorito. Estamos incumpliendo el SRP!
+
+#### Aggregates + VOs
+
+- Tendría más sentido que se llamara `PokemonFavourites` que `PokemonIdCollection`. Ya que el listado de entidades que
+  forma parte de un usuario son sus pokemon favoritos.
+
+#### No se trabajan con tipos primitivos en dominio
+
+OK
+
+#### Hay servicios de dominio
+
+- Se llama directamente al repository.
```

Comment:
Sigo sin ver el por qué un servicio de aplicación o caso de uso no puede llamar a un repositorio cuando la lógica es tan sencilla. No veo la necesidad de meter un servicio de dominio cuando se trabaja solamente con un repo. Ok, se puede encapsular luego eventos y demás... pero es que también lo puede hacer el caso de uso. Creo que meter tanta capa es uno de los problemas que encuentro en esto de la arquitectura hexagonal. Es mi sensación :D

Reply:
Aunque este punto es discutible, yo os doy herramientas de como teóricamente se explica que se puede hacer. En clase lo hemos comentado alguna vez. Yo lo que quiero es que experimentéis con estas herramientas y luego os forméis vuestras propias opiniones de qué os funciona mejor y qué os funciona peor. Yo en mi vida profesional he visto también el camino que comentas y tiene sus ventajas y sus inconvenientes. En este caso concreto, en clase os he insistido en que añadáis servicios de dominio porque lo más habitual es que no sean operaciones tan sencillas como buscar y añadir el pokemon favorito (que además no puedan tener cabida en agregados).